components/popover.tsx

```diff
@@ -0,0 +1,63 @@
+import type { ReactNode, FC, MouseEvent } from "react";
+import { useState } from "react";
+import Popover from "@mui/material/Popover";
+import Typography from "@mui/material/Typography";
+import Button from "@mui/material/Button";
+import Box from "@mui/material/Box";
+
+interface PopoverProps {
+  buttonContent: ReactNode;
+  popoverContent: ReactNode;
+}
+
+// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+export const HomePagePopover: FC<Readonly<PopoverProps>> = ({
+  buttonContent,
+  popoverContent,
+}: Readonly<PopoverProps>) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+
+  const handleClick = (event: MouseEvent<HTMLButtonElement>): void => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = (): void => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? "simple-popover" : undefined;
+
+  return (
+    <div>
+      <Button
+        aria-describedby={id}
+        variant="contained"
+        onClick={handleClick}
+        size="large"
+      >
+        {buttonContent}
+      </Button>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        onClose={handleClose}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "center",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "center",
+        }}
+      >
+        <Box maxWidth={400}>
+          <Typography sx={{ p: 2 }} align="center">
```

Comment:
Always prefer component props instead of `sx`. In this case it would be:
```suggestion
          <Typography p="2" align="center">
```
This is because component properties like `p` and `align` modify how the component functions from the inside, while `sx` just applies another layer of CSS on top

Reply:
That makes sense. However, changing the value to p="2" does not have the same desired padding effect it seems.